docs/source/cpp/api/flightsql.rst

```diff
@@ -0,0 +1,66 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: cpp
+.. highlight:: cpp
+
+================
+Arrow Flight SQL
+================
+
+.. note:: Flight SQL is currently experimental and APIs are subject to change.
+
+Common Types
+============
+
+.. doxygentypedef:: arrow::flight::sql::SqlInfoResult
+
+.. doxygentypedef:: arrow::flight::sql::SqlInfoResultMap
+
+.. doxygenenum:: arrow::flight::sql::SqlInfoOptions::SqlInfo
```

Comment:
Perhaps you want to define a doxygen group encompassing these typedefs and enums?